### PR TITLE
Reduce nested code in some functions

### PR DESCRIPTION
**Pull request by milmazz**

### CODE REVIEW

File: lib/hex/resolver/backtracks.ex [hunk 1]
```diff
@@ -93,19 +93,19 @@ defmodule Hex.Resolver.Backtracks do
...
-          case try_merge_package_version(list, version, parents) do
-            {:ok, list} ->
-              Map.put(map, name, list)
-            :error ->
-              Map.put(map, name, [{[version], parents}|list])
-          end
-
+         list
+         |> try_merge_package_version(version, parents)
+         |> process_merge_package_version(map, name, version, parents, list)
```
Comment (lexmag) [inline lib/hex/resolver/backtracks.ex]: I think previously code was clearer, new function takes too many arguments.

File: lib/hex/remote_converger.ex [hunk 2]
```diff
@@ -149,17 +149,17 @@ defmodule Hex.RemoteConverger do
...
-          retired ->
-            Hex.Shell.warn "  #{name} #{version} RETIRED!"
-            Hex.Shell.warn "    (#{retirement_reason(retired[:reason])}) #{retired[:message]}"
-        end
+        name |> Registry.retired(version) |> show_status(name, version)
       end)
     end
   end
 
+  defp show_status(nil, name, version), do: Hex.Shell.info IO.ANSI.format [:green, "  #{name} #{version}"]
```
Comment (lexmag) [inline lib/hex/remote_converger.ex]: We usually prefix such functions with `print_` instead of `show_`.

Comment (lexmag) [inline lib/hex/remote_converger.ex]: Also there is too much things are happening inside this clause, I think we should use `do ... end` syntax.

File: lib/hex/remote_converger.ex [hunk 3]
```diff
@@ -173,26 +173,31 @@ defmodule Hex.RemoteConverger do
...
+          verify_registry(deps, name, version, checksum)
         _ ->
           :ok
       end
     end)
   end
 
+  defp verify_registry(deps, name, version, checksum) do
+    registry_checksum = Registry.checksum(name, version)
+    if checksum && Base.decode16!(checksum, case: :lower) != registry_checksum,
```
Comment (lexmag) [inline lib/hex/remote_converger.ex]: `do ... end` syntax would be preferable here.

File: lib/hex/remote_converger.ex [hunk 4]
```diff
@@ -173,26 +173,31 @@ defmodule Hex.RemoteConverger do
...
+    if deps, do: verify_dependencies(deps, name, version)
+  end
+
+  defp verify_dependencies(deps, name, version) do
+    deps =
+      Enum.map(deps, fn {app, req, opts} ->
+        {Atom.to_string(opts[:hex]), Atom.to_string(app), req, !!opts[:optional]}
+      end)
+
+    if Enum.sort(deps) != Enum.sort(Registry.deps(name, version)),
```
Comment (lexmag) [inline lib/hex/remote_converger.ex]: `do ... end` syntax would be preferable here.

File: lib/hex/crypto/pbes2_hmac_sha2.ex [hunk 5]
```diff
@@ -58,6 +53,9 @@ defmodule Hex.Crypto.PBES2_HMAC_SHA2 do
   end
   def decrypt(_, _, _, _), do: :error
 
+  defp process_fetch_p2s({:ok, salt}, hash, passwd), do: {:ok, %{hash: hash, password: passwd}}
```
Comment (lexmag) [inline lib/hex/crypto/pbes2_hmac_sha2.ex]: Maybe `handle_p2s`? `process_fetch_p2s` seems misleading.

File: lib/hex/crypto/key_manager.ex [hunk 6]
```diff
@@ -28,12 +28,9 @@ defmodule Hex.Crypto.KeyManager do
         case module.init(protected, opts) do
           {:ok, params} ->
             key_manager = %KeyManager{module: module, params: params}
-            case ContentEncryptor.init(protected, opts) do
-              {:ok, content_encryptor} ->
-                {:ok, key_manager, content_encryptor}
-              content_encryptor_error ->
-                content_encryptor_error
-            end
+            protected
```
Comment (lexmag) [inline lib/hex/crypto/key_manager.ex]: We can probably have `fetch_content_encryptor` function instead:
```elixir
defp fetch_content_encryptor(key_manager, protected, opts) do
  case ContentEncryptor.init(protected, opts) do
    {:ok, content_encryptor} ->
      {:ok, key_manager, content_encryptor}
    error ->
      error
end
```

File: lib/hex/crypto/pbes2_hmac_sha2.ex (hunk 5 above)
Comment (milmazz) [inline lib/hex/crypto/pbes2_hmac_sha2.ex]: Thanks, I was looking for a better name in this function...

File: lib/hex/crypto/key_manager.ex (hunk 6 above)
Comment (milmazz) [inline lib/hex/crypto/key_manager.ex]: That also works, I'll change this section as you suggested.

**Comment by milmazz** [comment]
> my only minor comment is that we should have only one pipe per line

@ericmj Already take care of that in the latest commit.

@lexmag Thanks for your review, please let me know what do you think about the latest commit.

**Pull request merged**